README.md

```diff
@@ -26,69 +28,155 @@ Scheduler.start(&task);
 
 ```cpp
 class BlinkTask : public Task {
-protected:
-    void setup() {
-        state = HIGH;
+ protected:
+  void setup() {
+    state = HIGH;
 
-        pinMode(2, OUTPUT);
-        digitalWrite(2, state);
-    }
+    pinMode(2, OUTPUT);
+    digitalWrite(2, state);
+  }
 
-    void loop() {
-        state = state == HIGH ? LOW : HIGH;
-        digitalWrite(2, state);
+  void loop() {
+    state = state == HIGH ? LOW : HIGH;
+    digitalWrite(2, state);
 
-        delay(1000);
-    }
+    delay(1000);
+  }
 
 private:
-    uint8_t state;
+  uint8_t state;
 } blink_task;
 ```
 
-**IMPORTANT: Tasks must be declared globally on the stack (not a pointer). Failure to do so will crash your device.**
+**IMPORTANT:** Tasks must be declared globally on the stack (not a pointer). Failure to do so will crash your device.
 
-Tasks can run ```yield``` and ```delay``` like they normally would. These functions yield control to the scheduler
+Tasks can run ```yield()``` and ```delay()``` like they normally would. These functions yield control to the scheduler
 rather than the ESP8266.
 
-### Advanced Task Functions
+## Creating a LeanTask
 
-The ```Task``` also exposes a ```bool shouldRun()``` method that is used determine if the task loop
-should be be resumed. This can be inherited to add your own logic to determine if your code should be resumed.
+Lean tasks are classes that should inherit the ```LeanTask``` class. A lean task can define a ```loop()``` and ```setup()``` function
+much as the normal Arduino standard.
+
+LeanTask doesn't use ```cont.h``` (runs in the global context), so ```yield()``` works exactly as it would without using the library. This can be useful for saving ram (4k) if the task does not use ```yield()``` to interrupt the task. You can use ```delay()``` at the end of a task to set the interval.
 
+**Good example:**
 ```cpp
-bool shouldRun() {
-    bool run = Task::shouldRun();
+class MemTask : public LeanTask {
+ public:
+  void loop() {
+    Serial.print("Free Heap: ");
+    Serial.print(ESP.getFreeHeap());
+    Serial.println(" bytes");
+
+    delay(10000);
+  }
+} mem_task;
+```
 
-    // Your code here
+**Bad example. The first ```delay()``` does nothing:**
+<details><summary>Source. DO NOT DO THIS!</summary>
+<p>
+
+```cpp
+class PrintTask : public LeanTask {
+ protected:
+  void loop() {
+    Serial.println("Print Loop Start");
 
-    return run;
-}
+    delay(5000);
+
+    Serial.println("Print Loop End");
+
+    delay(5000);
+  }
+} print_task;
 ```
 
-**This function handles the ```delay()``` logic. The parent method should be called.**
+</p>
+</details>
 
-# Documentation
+Tasks can run ```yield()``` and ```delay()``` like they normally would. The ```yield()``` function transfer control to the ESP8266, NOT the scheduler. The ```delay()``` function will tell the scheduler that a delay is needed before the next run. If you REALLY need a delay, use ```::delay()```, but this will block the task and the scheduler.
```

Comment:
```suggestion
Tasks can run ```yield()``` and ```delay()``` like they normally would. The ```yield()``` function transfers control to the ESP8266, NOT the scheduler. The ```delay()``` function will tell the scheduler that a delay is needed before the next run. If you REALLY need a delay, use ```::delay()```, but this will block the task and the scheduler.
```